src/modules/src/controller_lee_payload.c

```diff
@@ -497,13 +468,19 @@ static struct vec computeDesiredVirtualInput(controllerLeePayload_t* self, const
 
   // push the latest change to the QP
   qpinput.F_d = F_d;
-  qpinput.M_d = M_d;
+  qpinput.M_d = M_d; 
   qpinput.plStPos = mkvec(state->payload_pos.x, state->payload_pos.y, state->payload_pos.z);
   qpinput.statePos = mkvec(state->position.x, state->position.y, state->position.z);
+
+  // We assume that we always have at least 1 neighbor
   qpinput.statePos2 = mkvec(state->neighbors[0].pos.x, state->neighbors[0].pos.y, state->neighbors[0].pos.z);
-  qpinput.statePos3 = mkvec(state->neighbors[1].pos.x, state->neighbors[1].pos.y, state->neighbors[1].pos.z);
   qpinput.ids[0] = state->neighbors[0].id;
-  qpinput.ids[1] = state->neighbors[1].id;
+  qpinput.num_neighbors = state->num_neighbors;
+  if (state->num_neighbors == 2) 
+  {
+    qpinput.statePos3 = mkvec(state->neighbors[1].pos.x, state->neighbors[1].pos.y, state->neighbors[1].pos.z);
```

Comment:
I would change qpinput.statePos to be an array (similar to the state in stablizer_types). Then, you can just use a simple loop here.